lib/ABI/FunctionType/ValueDistributor.h

```diff
@@ -85,6 +85,21 @@ class ValueDistributor {
   uint64_t LastAddedStackPadding = 0;
   uint64_t ArgumentIndex = 0;
 
+public:
+  ValueDistributor(const ValueDistributor &Another) = default;
+  ValueDistributor(ValueDistributor &&Another) = default;
+  ValueDistributor &operator=(const ValueDistributor &Another) {
+    revng_assert(&ABI == &Another.ABI);
```

Comment:
Because sometimes we want to revert the state (one of the big bugs this PR fixes). Are there more elegant solutions? Of course. Is this the simplest one? For sure!

Reply:
I think the risk of changing ABI is lower than breaking the copy constructor when we add a field to `ValueDistributor`, so I'd leave the default copy constructor.

But no biggie. Up to you.